instrumentation/opentelemetry-instrumentation-sklearn/src/opentelemetry/instrumentation/sklearn/__init__.py

```diff
@@ -0,0 +1,738 @@
+            )
+            delattr(delegator, "_otel_original_fn")
+        else:
+            logger.debug(
+                "Already uninstrumented: %s.%s", qualname, method_name,
+            )
+
+    def _uninstrument_instance_method(
+        self, estimator: BaseEstimator, method_name: str
+    ):
+        """Uninstrument an instance method.
+
+        Replaces the patched method with the original, and deletes the
+        attribute which stored the original method.
+
+        Args:
+            estimator (BaseEstimator): A class or instance of an ``sklearn``
+              estimator.
+            method_name (str): The method name of the estimator on which to
+              apply a span.
+       """
+        orig_method_name = "_otel_original_" + method_name
+        if isclass(estimator):
+            qualname = estimator.__qualname__
+        else:
+            qualname = estimator.__class__.__qualname__
+        if self._check_instrumented(estimator, method_name):
+            logger.debug(
+                "Uninstrumenting: %s.%s", qualname, method_name,
+            )
+            _, orig_method = getattr(estimator, orig_method_name)
+            setattr(
+                estimator, method_name, orig_method,
+            )
+            delattr(estimator, orig_method_name)
+        else:
+            logger.debug(
+                "Already uninstrumented: %s.%s", qualname, method_name,
+            )
+
+    def _instrument_class_method(
+        self, estimator: Type[BaseEstimator], method_name: str
+    ):
+        """Instrument an estimator method with a span.
+
+        When instrumenting we attach a tuple of (Class, method) to the
+        attribute ``_original_<method_name>`` for each method. This allows
+        us to replace the patched with the original in unstrumentation, but
+        also allows proper instrumentation of child classes without
+        instrumenting inherited methods twice.
+
+        Args:
+            estimator (BaseEstimator): A ``BaseEstimator``-derived
+              class
+            method_name (str): The method name of the estimator on which to
+              apply a span.
+        """
+        if self._check_instrumented(estimator, method_name):
+            logger.debug(
+                "Already instrumented: %s.%s",
+                estimator.__qualname__,
+                method_name,
+            )
+            return
+        class_attr = getattr(estimator, method_name)
+        delegator = get_delegator(estimator, method_name)
+        if isinstance(class_attr, property):
+            logger.debug(
+                "Not instrumenting found property: %s.%s",
+                estimator.__qualname__,
+                method_name,
+            )
+        elif delegator is not None:
+            implement_spans_delegator(delegator)
+        else:
+            setattr(
+                estimator,
+                "_otel_original_" + method_name,
+                (estimator, class_attr),
```

Comment:
Just wondering why the original needs to be a `tuple` instead of just the `class_attr`?

Reply:
From above: https://github.com/open-telemetry/opentelemetry-python/pull/1054#discussion_r517376055

> We want to make sure that we uninstrument properly in child classes by replacing patches with originals, only if the method is uninherited.